Reconstruction/RecTracker/src/components/ExtrapolationTest.cpp

```diff
@@ -0,0 +1,138 @@
+
+#include "DetInterface/IGeoSvc.h"
+#include "DetInterface/ITrackingGeoSvc.h"
+#include "RecInterface/ITrackSeedingTool.h"
+
+#include "GaudiKernel/IRndmGenSvc.h"
+
+
+#include "ACTS/Detector/TrackingGeometry.hpp"
+#include "ACTS/EventData/Measurement.hpp"
+#include "ACTS/Examples/BuildGenericDetector.hpp"
+#include "ACTS/Extrapolation/ExtrapolationCell.hpp"
+#include "ACTS/Extrapolation/ExtrapolationEngine.hpp"
+#include "ACTS/Extrapolation/IExtrapolationEngine.hpp"
+#include "ACTS/Extrapolation/MaterialEffectsEngine.hpp"
+#include "ACTS/Extrapolation/RungeKuttaEngine.hpp"
+#include "ACTS/Extrapolation/StaticEngine.hpp"
+#include "ACTS/Extrapolation/StaticNavigationEngine.hpp"
+#include "ACTS/Fitter/KalmanFitter.hpp"
+#include "ACTS/Fitter/KalmanUpdator.hpp"
+#include "ACTS/MagneticField/ConstantBField.hpp"
+#include "ACTS/Surfaces/PerigeeSurface.hpp"
+#include "ACTS/Utilities/Definitions.hpp"
+#include "ACTS/Utilities/Identifier.hpp"
+#include "ACTS/Utilities/Logger.hpp"
+
+#include "datamodel/PositionedTrackHitCollection.h"
+#include "datamodel/TrackHitCollection.h"
+
+#include "DD4hep/LCDD.h"
+#include "DD4hep/Volumes.h"
+#include "DDRec/API/IDDecoder.h"
+#include "DDSegmentation/BitField64.h"
+
+#include <cmath>
+#include <random>
+
+#include "ExtrapolationTest.h"
+
+DECLARE_ALGORITHM_FACTORY(ExtrapolationTest)
+
+ExtrapolationTest::ExtrapolationTest(const std::string& name, ISvcLocator* svcLoc) : GaudiAlgorithm(name, svcLoc) {
+
+  declareProperty("positionedTrackHits", m_positionedTrackHits, "hits/TrackerPositionedHits");
+}
+
+ExtrapolationTest::~ExtrapolationTest() {}
+
+StatusCode ExtrapolationTest::initialize() {
+
+  IRndmGenSvc* randSvc = svc<IRndmGenSvc>("RndmGenSvc", true);
+
+  m_geoSvc = service("GeoSvc");
+
+  StatusCode sc = GaudiAlgorithm::initialize();
+  if (sc.isFailure()) return sc;
+
+  m_trkGeoSvc = service("TrackingGeoSvc");
+  if (nullptr == m_trkGeoSvc) {
+    error() << "Unable to locate Tracking Geometry Service. " << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  m_trkGeo = m_trkGeoSvc->trackingGeometry();
+  auto propConfig = RungeKuttaEngine<>::Config();
+  propConfig.fieldService = std::make_shared<ConstantBField>(0, 0, m_magneticFieldBz * 0.001); // needs to be in kT
```

Comment:
Can we somehow use one of our units? This looks like it's going to cause confusion / bugs :)